spec/game_spec.rb

```diff
@@ -1,17 +1,125 @@
+    end
   end
-  context " plus-minus factoring for user_input" do
-    it "check class of user input"
-    it "when answer_plus - array is created"
-    it "when answer_minus -hash is created"
+
+  context 'when the method plus-minus factoring output correct value' do
+    [
+      [[6, 5, 4, 1], [6, 5, 4, 1], '++++'],
+      [[1, 5, 3, 2], [5, 1, 3, 2], '++--'],
+      [[1, 2, 3, 4], [1, 3, 2, 4], '++--'],
+      [[1, 2, 3, 4], [1, 2, 4, 3], '++--'],
+      [[1, 2, 3, 4], [1, 4, 2, 3], '+---'],
+      [[1, 2, 3, 4], [4, 2, 1, 3], '+---'],
+      [[1, 2, 3, 4], [2, 4, 3, 1], '+---'],
+      [[1, 2, 3, 4], [2, 3, 1, 4], '+---'],
+      [[1, 2, 3, 4], [4, 3, 2, 1], '----'],
+      [[5, 4, 3, 2], [2, 3, 4, 5], '----'],
+      [[1, 2, 3, 4], [2, 1, 4, 3], '----'],
+      [[1, 3, 4, 2], [1, 2, 3, 4], '+---'],
+      [[5, 2, 5, 5], [2, 5, 5, 5], '++--'],
+      [[5, 5, 2, 5], [2, 5, 5, 5], '++--'],
+      [[5, 5, 5, 2], [2, 5, 5, 5], '++--'],
+      [[6, 2, 6, 2], [2, 6, 2, 6], '----'],
+      [[6, 6, 2, 2], [2, 6, 2, 6], '++--'],
+      [[2, 2, 6, 6], [2, 6, 2, 6], '++--'],
+      [[2, 6, 6, 2], [2, 6, 2, 6], '++--'],
+      [[6, 2, 2, 6], [2, 6, 2, 6], '++--'],
+      [[3, 1, 3, 5], [3, 3, 1, 5], '++--'],
+      [[3, 5, 1, 3], [3, 3, 1, 5], '++--'],
+      [[3, 3, 5, 1], [3, 3, 1, 5], '++--'],
+      [[1, 3, 5, 3], [3, 3, 1, 5], '+---'],
+      [[5, 3, 1, 3], [3, 3, 1, 5], '++--'],
+      [[1, 5, 3, 3], [3, 3, 1, 5], '----'],
+      [[5, 3, 3, 1], [3, 3, 1, 5], '+---'],
+      [[1, 2, 3, 4], [4, 3, 4, 3], '--']
+    ].each do |item|
+      it "when secret_code is #{item[0]} && the user input is #{item[1]}, the responds to consol will be #{item[2]}" do
+        allow(game).to receive(:plus_minus_factoring).and_call_original
+        game.instance_variable_set(:@secret_code, item[0])
+        expect(game.plus_minus_factoring(item[1].join) == item[2]).to eq(false)
+        game.plus_minus_factoring(item[1].join)
+      end
+    end
+  end
+
+  context 'when user get the hint' do
+    it 'when hint exits' do
+      allow(game).to receive(:hint).and_call_original
+      expect(game.hint).to be_instance_of(String)
+    end
+    it 'when hint is only 1 number' do # :TODO don't pass evry time!
+      allow(game).to receive(:hint).and_call_original
+      expect(game.hint.length).to eq(1)
+    end
+    it 'when  reminder after first hint is exists' do
+      allow(game).to receive(:hint).once.and_call_original
+      expect(game.hint_clone_scode).to be_instance_of(Array)
+    end
+  end
+
+  context 'when the hit was viewed' do
+    before do
+      expect(game).to receive(:hint).exactly(HINT - HINT + 1).times # :TODO How to get result of method implementation N-times?
+    end
+
+    before do
+      expect(game).to receive(:hint).exactly(HINT - HINT + 2).times
+    end
+
+    it 'when  reminder after first hint is correct' do
+      # allow(game).to receive(:hint).and_call_original
+      expect(game.hint_clone_scode.size).to eq(DIGIT - 1)
```

Comment:
7) Game when the hit was viewed when  reminder after first hint is correct
     Failure/Error: expect(game.hint_clone_scode.size).to eq(DIGIT - 1)
     
       expected: 3
            got: 4
     
       (compared using ==)